packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -262,28 +262,60 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
      * This is the generic `Open` method. Opens files and directories too. Resolves to the opened URI.
      * Except when you are on either Windows or Linux `AND` running in electron. If so, it opens a file.
      */
-    protected async doOpen(): Promise<URI | undefined> {
+    protected async doOpen(): Promise<MaybeArray<URI> | undefined> {
         if (!isOSX && this.isElectron()) {
             return this.doOpenFile();
         }
         const [rootStat] = await this.workspaceService.roots;
         const destinationUri = await this.fileDialogService.showOpenDialog({
```

Comment:
`destinationUri` is no longer a good name for what might be multiple URI's. `selectedUris` comes to mind.

Reply:
Changed to selectedUris and selectedFilesUris.